exportReport/Export/ExportReport.java

```diff
@@ -0,0 +1,255 @@
+							@Override
+							public void propertyChange(PropertyChangeEvent evt) {
+								ReportFilter filter = (ReportFilter) evt.getNewValue();
+								String extension = (filter.getExtensionByDescription(filter.getDescription())
+										.length() == 0 ? ""
+												: "." + filter.getExtensionByDescription(filter.getDescription()));
+								fc.setSelectedFile(new File(extension));
+							}
+						});
+			}
+
+			int rc = fc.showSaveDialog(View.getSingleton().getMainFrame());
+
+			boolean bool = false;
+			String fileExtension = "";
+			while (rc == WritableFileChooser.APPROVE_OPTION && !bool) {
+				fileExtension = list.compareExtension(fc.getSelectedFile().getName());
+				if (fileExtension.length() > 0) {
+					bool = true;
+				}
+				if (!bool) {
+					JOptionPane.showMessageDialog(null,
+							"The file " + fc.getSelectedFile() + " is not a valid destination file.", "Open Error",
+							JOptionPane.ERROR_MESSAGE);
+					rc = fc.showSaveDialog(View.getSingleton().getMainFrame());
+				}
+			}
+
+			if (rc == WritableFileChooser.APPROVE_OPTION & bool) {
+				task = new Task(view, extension, fileExtension);
+				task.execute();
+			}
+
+		} catch (Exception e) {
+			logger.error(e.getMessage(), e);
+			view.showWarningDialog("Error occured, please see log for further details.");
+		}
+	}
+
+	private FileList generateFileList() {
+		FileList list = new FileList();
+		list.add(Utils.html, Utils.htmlType, Utils.html, Utils.htmlDescription, Utils.htmlIcon);
+		list.add(Utils.bootstrap, Utils.bootstrapType, Utils.bootstrap, Utils.bootstrapDescription,
+				Utils.bootstrapIcon);
+		list.add(Utils.xml, Utils.xmlType, Utils.xml, Utils.xmlDescription, Utils.xmlIcon);
+		list.add(Utils.json, Utils.jsonType, Utils.json, Utils.jsonDescription, Utils.jsonIcon);
+		list.add(Utils.pdf, Utils.pdfType, Utils.pdf, Utils.pdfDescription, Utils.pdfIcon);
+		list.add(Utils.doc, Utils.docType, Utils.doc, Utils.docDescription, Utils.docIcon);
+		return list;
+	}
+
+	private WritableFileChooser generateWriteableFileChooser(WritableFileChooser fc, FileList list) {
+		fc = new WritableFileChooser(Model.getSingleton().getOptionsParam().getUserDirectory());
+
+		fc.addChoosableFileFilter(new ReportFilter(list, Utils.ALL));
+		fc.addChoosableFileFilter(new ReportFilter(list, Utils.html));
+		fc.addChoosableFileFilter(new ReportFilter(list, Utils.bootstrap));
+		fc.addChoosableFileFilter(new ReportFilter(list, Utils.xml));
+		fc.addChoosableFileFilter(new ReportFilter(list, Utils.json));
+		fc.addChoosableFileFilter(new ReportFilter(list, Utils.pdf));
+		fc.addChoosableFileFilter(new ReportFilter(list, Utils.doc));
+
+		fc.setAcceptAllFileFilterUsed(false);
+
+		fc.setFileView(new ReportFileView(list));
+
+		return fc;
+	}
+
+	private void deleteFile(String str, String err) {
+		File f = new File(str);
+		boolean deleted = false;
+		if (f.exists() && !f.isDirectory()) {
+			deleted = f.delete();
+		}
+		if (!deleted) {
+			// Print to err
```

Comment:
Done. Message to logger saying the file could not be removed, also considering a gui message but deferring that till next update or until feedback on usage of the message dialog.